Release pixelator 1.3.0 (#106)
nf-core/pixelator: docs/output.md

</details>

### Quality control

Quality control is performed using `pixelator single-cell preqc` and `pixelator single-cell adapterqc`.

> The preqc stage performs QC and quality filtering of the raw sequencing data.

Reviewer: ```suggestion
The preqc step performs QC and quality filtering of the raw sequencing data using [Fastp](https://github.com/OpenGene/fastp) internally.
```
I think

> It also generates a QC report in HTML and JSON formats. It saves processed reads as well as reads that were

Reviewer: ```suggestion
It generates a QC report in HTML and JSON formats. It saves processed reads as well as reads that were
```

discarded (i.e. were too short, had too many Ns, or too low quality, etc.). Internally `preqc`

> uses [Fastp](https://github.com/OpenGene/fastp), and `adapterqc`
> uses [Cutadapt](https://cutadapt.readthedocs.io/en/stable/).

Reviewer: ```suggestion
```
moved to their respective sections

> The `adapterqc` stage checks for the presence and correctness of the pixel binding sequences.

Reviewer: ```suggestion
The `adapterqc` stage checks for the presence and correctness of the pixel binding sequences, 
using [Cutadapt](https://cutadapt.readthedocs.io/en/stable/) internally.
```

It also generates a QC report in JSON format. It saves processed reads as well as discarded reads (i.e. reads that did not have a match for both pixel binding sequences).

These processed and discarded FASTQ reads are intermediate and by default not placed in the output folder with the final files delivered to users.
Set `--save_qc_passed_reads` and/or `--save_qc_passed_reads` to enable publishing of these files.
Alternatively, set `--save_all` to keep all intermediary outputs of all steps.

<details markdown="1">

> <summary>Output files</summary>

Reviewer: HTML outputs are missing in the list of outputs

Author: Good catch, thanks! The html reports were missing from the output folder as well. Fixed now.

</details>

### Compute connected components

This step uses the `pixelator single-cell graph` command.
The input is the edge list parquet file generated in the collapse step.
The molecules from edge list are filtered by count (`--graph_min_count`) to form the edges of the connected components of the graph.
When graphs are computed and identified, their ID names are added back to the edge list in a column called "component".

The graph command has the option to recover components (technical multiplets) into smaller
components using community detection to find and remove problematic edges
(see `--multiplet_recovery`). These new component IDs are then stored in the "component" column. The information to keep track of the original and
newly recovered components are stored in a file (components_recovered.csv).
This file is not included in the output folder by default, but can be included by passing `--save_recovered_components`.

> The edgelist is intermediate and by default not placed in the output folder with the final files delivered to users.

Reviewer: ```suggestion
The edge list is intermediate and by default not placed in the output folder with the final files delivered to users.
```

Reviewer: inconsist spelling of this throughout this document, please settle on one spelling 🙂

(AnnData object)[https://anndata.readthedocs.io/en/latest/] as well as some useful metadata.

Some summary statistics before filtering are stored in `raw_components_metrics.csv.gz`.
This file is not included in the output folder by default, but can be included by passing `--save_raw_component_metrics`.

By default, the PXL file after annotate will not be saved to the results directory unless `--skip_analysis` and `--skip_layout` is passed.
Set `--save_annotate_dataset` to include these files.

<details markdown="1">
<summary>Output files</summary>

- `pixelator`

  - `annotate`
    - `<sample-id>.annotate.dataset.pxl`: The annotated PXL dataset,
    - `<sample-id>.meta.json`: Command invocation metadata.
    - `<sample-id>.raw_components_metrics.csv.gz`
    - `<sample-id>.report.json`: Statistics for the analysis step.
  - `logs`
    - `<sample-id>.pixelator-annotate.log`: pixelator log output.
    </details>

### Downstream analysis

This step uses the `pixelator single-cell analysis` command.
Downstream analyses are performed on the PXL file generated by the previous stage.
The results of the analysis are added to the PXL file produced in this stage.

> Currently, the following analysis are performed:

Reviewer: ```suggestion
Currently, the following analyses are performed:
```
